client/src/pages/LyricsView/LyricsDisplay.tsx

```diff
@@ -7,32 +7,58 @@ import {
   InputAdornment,
   Link,
   makeStyles,
+  Toolbar,
   TextField,
   Typography
 } from "@material-ui/core";
 import SearchIcon from "@material-ui/icons/Search";
 import CloseIcon from "@material-ui/icons/Close";
 
+import SyncEnabledIcon from "@material-ui/icons/Sync";
+import SyncDisabledIcon from "@material-ui/icons/SyncDisabled";
+
 import "./LyricsDisplay.css";
+import { Sync } from "@material-ui/icons";
+
+// adjusting for latency to highlight lyrics due to the time it takes to render the components on screen
+const LATENCY_ADJUSTMENT_MAGIC_VALUE: number = 135;
+
+interface ILRCContent {
+  content: string;
+  timestamp: number;
+}
 
 interface IProps {
   lyrics?: string;
+  syncedLyricsArray?: Array<ILRCContent>;
   lyricsArtist?: string;
   lyricsTitle?: string;
   geniusUrl?: string;
+  progressMs?: number;
 }
 
 const LyricsDisplay: React.FunctionComponent<IProps> = ({
+  syncedLyricsArray,
   lyrics,
-  lyricsArtist,
-  lyricsTitle,
-  geniusUrl
+  progressMs
 }) => {
+  //cloning the original lyrics array sp that we can restore the lyrics if a user rewinds the song
+  const clonedLyricsArray = syncedLyricsArray === undefined ? [] : [...syncedLyricsArray];
   const classes = useStyles();
   const lyricsRef = useRef<HTMLDivElement | null>(null);
   const searchInputRef = useRef<HTMLInputElement | null>(null);
+  const highlightedRef = useRef<HTMLSpanElement | null>(null);
   const [search, setSearch] = useState("");
   const [searchShown, setSearchShown] = useState(false);
+  const [syncedLyrics, setSyncedLyrics] = useState({
+    array: clonedLyricsArray,
+    before: "",
+    highlighted: "",
+    after: lyrics
+  });
+  const [naturalSongProgress, setNaturalSongProgress] = useState(0);
+  const [syncEnabled, setSyncEnabled] = useState(true);
```

Comment:
I found that when disabling the sync - it still auto-scrolled me to the last position. Could we also make all the text go to the normal size when sync is disabled?

Also when non-time synced lyrics appeared, the toolbar went up high (search input goes under the top toolbar) and there was a lot of whitespace at the top of the text (the empty highlighted element was doing this)

Reply:
The whitespaces I had already fixed by removing the <br> on html if the sync was not possible.

Not sure I understood entirely the portion about scrolling to the last position, or if I could reproduce it, but fixed the lyrics size also!